src/gui/attributetable/qgsattributetablemodel.h

```diff
@@ -395,6 +395,12 @@ class GUI_EXPORT QgsAttributeTableModel: public QAbstractTableModel
     //! Changed attribute values within a bulk edit command
     QMap<QPair<QgsFeatureId, int>, QVariant> mAttributeValueChanges;
 
+    //! Inserted feature IDs within a bulk edit command
+    QList<QgsFeatureId> mInsertedRowsChanges;
+
+    //! TRUE if triggered by afterRollback()
+    bool mIsRollingBack = false;
```

Comment:
The naming of this variable is not intuitive. "is rolling back" means we're in the middle of a rollback, so calling it from a signal which is sent *after* rolling back makes it counter-intuitive.